corehq/apps/app_manager/app_translations/generators.py

```diff
@@ -212,9 +198,15 @@ def __init__(self, translations, metadata):
             self._generate_translation_files(translations, metadata)
         except Exception:
             exc_info = sys.exc_info()
-            self.cleanup()
+            self._cleanup()
             six.reraise(*exc_info)
```

Comment:
You can restructure this context manager a bit to move the work out of the constructor.  The nice thing about that approach is that you could then drop this try/except/cleanup stuff, since `__exit__` will take care of that.

Specifically, you could drop `self.generated_files` entirely and make `generate_translation_files` public and call that in the client code (I'd save `translations` and `metadata` on the class so you don't need to pass them twice).  You could memoize `generate_translation_files` or make it a memoized property if it needs to be accessed more than once.

Reply:
Thanks @esoergel . Made all the changes.

Reply:
sorry, made one comment earlier. Looked like it didnt reach Github.

> You could memoize generate_translation_files or make it a memoized property if it needs to be accessed more than once.

Considering the changes done now, i don't see why one would need to access generated files more than once. Plus they can't anyway since it gets cleaned up. So don't think making it as `memoized` property is needed.